src/object_store/src/object/mod.rs

```diff
@@ -97,13 +101,46 @@ impl BlockLocation {
     }
 }
 
+#[async_trait::async_trait]
+pub trait StreamingUploader: Send {
+    fn write_bytes(&mut self, data: Bytes) -> ObjectResult<()>;
+
+    async fn finish(self) -> ObjectResult<()>;
+}
+
+pub enum StreamingUploaderImpl {
```

Comment:
`ObjectStoreImpl` currently implements `ObjectStore`, so if we want to return `MonitoredStreamingUploader`, we need to wrap it again in `Box`. One way to solve it would be to remove `impl ObjectStore` for `ObjectStoreImpl`.

Reply:
> One way to solve it would be to remove impl `ObjectStore` for `ObjectStoreImpl`.
I'm adopting this solution for now.